options.go

```diff
@@ -0,0 +1,61 @@
+package autonat
+
+import (
+	"errors"
+	"time"
+)
+
+// config holds configurable options for the autonat subsystem.
+type config struct {
+	getAddressFunc  GetAddrs
+	bootDelay       time.Duration
+	retryInterval   time.Duration
+	refreshInterval time.Duration
+	requestTimeout  time.Duration
+}
+
+var defaults = func(c *config) error {
+	c.bootDelay = 15 * time.Second
+	c.retryInterval = 90 * time.Second
+	c.refreshInterval = 15 * time.Minute
+	c.requestTimeout = 30 * time.Second
+
+	return nil
+}
+
+// WithAddresses allows overriding which Addresses the AutoNAT client beliieves
+// are "its own". Useful for testing, or for more exotic port-forwarding
+// scenarios where the host may be listening on different ports than it wants
+// to externally advertise or verify connectability on.
+func WithAddresses(addrFunc GetAddrs) Option {
```

Comment:
Maybe call this `AddressGuesser`? As far as I can tell, that's _really_ what this is.